src/client/dfs/dfs.c

```diff
@@ -4939,6 +4939,239 @@ dfs_ostat(dfs_t *dfs, dfs_obj_t *obj, struct stat *stbuf)
 	return rc;
 }
 
+struct dfs_statx_args {
+	dfs_t			*dfs;
+	dfs_obj_t		*obj;
+	struct stat		*stbuf;
+	daos_handle_t		parent_oh;
+};
+
+struct statx_op_args {
+	daos_key_t		dkey;
+	daos_iod_t		iod;
+	daos_recx_t		recx;
+	d_sg_list_t		sgl;
+	d_iov_t			sg_iovs[INODE_AKEYS];
+	struct dfs_entry	entry;
+	daos_array_stbuf_t	array_stbuf;
+};
+
+int
+ostatx_cb(tse_task_t *task, void *data)
+{
+	struct dfs_statx_args	*args = daos_task_get_args(task);
+	struct statx_op_args	*op_args = *((struct statx_op_args **)data);
+	int			rc2, rc = daos_der2errno(task->dt_result);
+
+	if (rc != 0) {
+		D_ERROR("Failed to stat file "DF_RC"\n", DP_RC(rc));
+		D_GOTO(out, rc);
+	}
```

Comment:
I think this is wrong, I expect rc is a daos errno here and it needs to be to use DF_RC however this function returns a system error number in other cases.  That said, I'm not sure where the RC of this function is used/passed to?

Reply:
well i had that right the first time around (for the printf) until you recommended the change in the dev channel :-/

otherwise this is using a system errno which is set in the event error completion or returned as the function return code if ev is NULL

Reply:
OK should be fixed now. what was missing is setting the task result from der to errno in this case because even if the CB return errno, tse will use the original task error from DAOS over the one from CB since it keeps the original failure code (in case there is a failure).
I added 2 test cases, one that exercises this code (error from DAOS, and the second that exercises the error next in this CB, to verify it returns appropriately the error.

Reply:
@ashleypittman any more comments on this?